app/src/main/java/com/aconno/sensorics/BluetoothScanningService.kt

```diff
@@ -135,23 +136,51 @@ class BluetoothScanningService : DaggerService() {
                     .subscribeOn(Schedulers.io())
                     .observeOn(AndroidSchedulers.mainThread())
                     .subscribe {
-                        bluetooth.startScanning(it)
-                        running = true
-                        startRecording()
-                        startLogging()
-                        startSyncing()
-                        handleInputsForActions()
+                        startScan(it)
                     }
             )
         } else {
+            startScan()
+        }
+
+        return START_STICKY
+    }
+
+
+    /**
+     * Restart scanning before Android BLE Scanning Timeout
+     * https://github.com/aconno/Sensorics/issues/12
+     */
+    private fun startTimer(deviceList: List<Device>? = null) {
+        //Launches non-blocking coroutine
+        scanTimerDisposable = GlobalScope.launch(context = Dispatchers.Main) {
+            delay(ANDROID_N_MAX_SCAN_DURATION - 60 * 1000)
+            Timber.tag("Sensorics - BLE").d("Stopping")
+            stopScanning()
+            Timber.tag("Sensorics - BLE").d("Restarting")
+            startScan(deviceList)
+        }
+    }
+
+    private fun startScan(deviceList: List<Device>? = null) {
+        startTimer(deviceList)
+        Timber.tag("Sensorics - BLE").d("Started")
+
+        if (deviceList == null) {
             bluetooth.startScanning()
             running = true
             startRecording()
             startLogging()
             startSyncing()
             handleInputsForActions()
+        } else {
+            bluetooth.startScanning(deviceList)
+            running = true
```

Comment:
Duplication of code. Move this out of the if/else statement.

Reply:
Removed Duplicate code